### DISCUSSION TITLE
addBulk idles worker, and put everything on waiting list

### DISCUSSION THREAD

**Discussion by madsenmm** [taskforcesh/bullmq#2338] [Q&A]
Why does it behave like this, I don't quite understand.
When using the `addBulk` method, all jobs are put on the waiting list, and the worker just goes to idle state, and never picking up any jobs?

What am I missing here 👀

**Comment by manast** [comment]
Do you have a test code that reproduces this behaviour?

**Reply by madsenmm** [reply]
Forgot to add to previous description: Works fine when running locally, only once its deployed they gut stuck on waiting list, with the one item in the active queue just sitting there 🤷 

Package versions:
```yml
"bullmq": "^4.14.0",
"ioredis": "^5.3.2",
```

_And running on Redis: v6.2.6, standalone mode._

Here is an example of how to reproduce it.
```js
// file to collect jobs
const updateQueue = new Queue('updates', {
  connection: 'redis-connection-url',
  defaultJobOptions: {
    removeOnComplete: true,
    removeOnFail: true,
    attempts: 2,
    backoff: {
      type: 'exponential',
      delay: 1000,
    },
  },
});

// Example of random jobs
const jobs = [];

for (let i = 0; i < 10; i += 1) {
  jobs.push({
    name: 'update',
    data: {
      test: `test-${i}`,
    },
  });
}

updateQueue.addBulk(jobs);

// instrumentation.ts (specific use for vercel, handles the worker)
const settings = {
  connection: 'redis-connection-url',
  concurrency: 3,
  removeOnComplete: {
    count: 1500,
  },
  removeOnFail: {
    count: 7500,
  },
  metrics: {
    maxDataPoints: MetricsTime.ONE_WEEK,
  },
} satisfies WorkerOptions;

const updateJob = async (job: Job) => {
  await new Promise((resolve) => setTimeout(resolve, 2000));
  return 'Job completed';
};

const worker = new Worker('updates', updateJob, settings);
```

**Reply by manast** [reply]
Works as expected, just tried with this exact code:
```js
const { Queue, Worker, MetricsTime } = require("bullmq");

// file to collect jobs
const updateQueue = new Queue("updates", {
  connection: "redis-connection-url",
  defaultJobOptions: {
    removeOnComplete: true,
    removeOnFail: true,
    attempts: 2,
    backoff: {
      type: "exponential",
      delay: 1000,
    },
  },
});

// Example of random jobs
const jobs = [];

for (let i = 0; i < 10; i += 1) {
  jobs.push({
    name: "update",
    data: {
      test: `test-${i}`,
    },
  });
}

updateQueue.addBulk(jobs);

// instrumentation.ts (specific use for vercel, handles the worker)
const settings = {
  connection: "redis-connection-url",
  concurrency: 3,
  removeOnComplete: {
    count: 1500,
  },
  removeOnFail: {
    count: 7500,
  },
  metrics: {
    maxDataPoints: MetricsTime.ONE_WEEK,
  },
};

const updateJob = async (job) => {
  console.log("processing job...", job.name);
  await new Promise((resolve) => setTimeout(resolve, 2000));
  return "Job completed";
};

const worker = new Worker("updates", updateJob, settings);

```

yields:
```bash
$ node 2338.js
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
processing job... update
```

**Reply by madsenmm** [reply]
On a local redis instance?
I get the same when running locally, and it also completes the list. But once I use the external redis, all jobs just stands in the waiting queue, with one item in active state.

I read the [connection](https://docs.bullmq.io/guide/connections) documentation, and doesn't seem any specific configuration to work as expected, besides the `maxmemory-policy=noeviction` setting

**Reply by manast** [reply]
Should not matter which Redis instance you use as long as it is a 100% compatible Redis instance.

**Reply by madsenmm** [reply]
Created on redis.com, so I would assume 😄 
It works as expected if I add each job separately, it only stops working once I use the `addBulk` method.
And not maxing out connections or database space either... 🤔

**Reply by manast** [reply]
If the code is exactly the same one, I find it difficult to believe that the fact that is running in a hosted Redis instance instead of locally can have something to do with it.

**Reply by madsenmm** [reply]
I'm kind of lost why it behaves as it does :/

**Reply by manast** [reply]
You can try using ```redis-cli monitor``` and compare the output from local to hosted, however you must isolate the issue in as little amount of operations as possible as the output can be quite verbose otherwise.

**Reply by manast** [reply]
You should be able to see the commands that are putting the jobs in wait and then moved to active by the workers.

**Reply by madsenmm** [reply]
I noticed that the the hosted Redis instance, starts with this in the monitor logs:
The message for the eval is way longer than what I've written here.
**Can this cause any issues?**

Consider i'm a complete noob on Redis and all this stuff, so the most basic stuff might be a possible error on my behalf 🙃 

```
"auth" "(redacted)" "(redacted)"
"info"
"eval" "--[[\n Move job from active to a finished status (completed o failed)\n A job can only be moved to completed if it was active.\n The job must be locked before it can be moved to a finished status,\n and the lock must be released in this script.\n Input:\n KEYS[1] wait key\n KEYS[2] active key\n KEYS[3] prioritized key\n KEYS[4] event stream key\n KEYS[5] stalled key\n -- Rate limiting\n KEYS[6] rate limiter key\n KEYS[7] delayed key\n KEYS[8] paused key\n KEYS[9] meta key\n KEYS[10] pc priority counter\n KEYS[11]...
```

**Reply by manast** [reply]
Thats normal, the first time a lua script is called it needs to send the whole script.
